contracts/Exchange.sol

```diff
@@ -154,22 +152,30 @@ contract Exchange is Restricted {
         return(activeBuyOrders.length, activeSellOrders.length);
     }
 
-    // returns CHUNK_SIZE orders starting from offset
+    // returns <chunkSize> active buy orders starting from <offset>
     // orders are encoded as [id, maker, price, amount]
-    function getActiveBuyOrders(uint offset) external view returns (uint[4][CHUNK_SIZE] response) {
-        for (uint8 i = 0; i < CHUNK_SIZE && i + offset < activeBuyOrders.length; i++) {
+    function getActiveBuyOrders(uint offset, uint16 chunkSize)
+    external view returns (uint[4][]) {
+        uint[4][] memory response = new uint[4][](chunkSize);
+        for (uint16 i = 0; i < chunkSize && i + offset < activeBuyOrders.length; i++) {
             uint64 orderId = activeBuyOrders[offset + i];
```

Comment:
Not that it matters much for these functions, but it would nicer and cheaper to iterate on the actual index (i.e. from `offset` to `min(arr.length, offset+chunkSize)`, precomputed)

Reply:
I agree, will do it.